### ISSUE TITLE
bug in to_snake alias generator in pydantic V2.8

### ISSUE THREAD

**Issue by simonarcher99** [pydantic/pydantic#9811]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

We've noticed a change of behaviour in pydantic v2.8, released today, that we think is an unintended change.

Previously when calling `to_snake` on "SomethingV2" the output would be "something_v2". Since pydantic v2.8 this appears to have started returning something_v_2.

I'm happy to put a fix in for this is someone can confirm that it was unintended and could point me in the right direction! Thanks

### Example Code

```Python
from pydantic.alias_generators import to_snake

pascal_case_string = "SomethingV2"

to_snake(pascal_case_string) # returns "something_v_2" when "something_v2" is expected
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.8.0
pydantic-core version: 2.20.0
pydantic-core build: profile=release pgo=true
install path: /Users/simon.archer/.pyenv/versions/3.11.9/envs/domestic-optimisation-3.11/lib/python3.11/site-packages/pydantic
python version: 3.11.9 (main, May 29 2024, 12:45:38) [Clang 15.0.0 (clang-1500.3.9.4)]
platform: macOS-14.5-arm64-arm-64bit
related packages: pydantic-settings-2.3.3 typing_extensions-4.12.2
commit: unknown
```

**Comment by sydney-runkle** [comment]
@simonarcher99,

Thanks for reporting this! Sorry about that. Looks like I broke this in https://github.com/pydantic/pydantic/pull/9561.

I think this is something we want to fix.

**Comment by sydney-runkle** [comment]
I'll open a PR shortly!

**Comment by simonarcher99** [comment]
Amazing thanks!

**Comment by sydney-runkle** [comment]
Fixed via the above PR, you can pull from `main` for the fix now, and we'll release v2.8.1 later this week!

**Comment by simonarcher99** [comment]
Perfect, thanks for fixing so quickly
